src/fragments/lib/auth/js/switch-auth.mdx

```diff
@@ -49,62 +49,164 @@ For more information about working with Lambda Triggers for custom authenticatio
 
 To initiate a custom authentication flow in your app, call `signIn` without a password. A custom challenge needs to be answered using the `sendCustomChallengeAnswer` method:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { CognitoUser } from '@aws-amplify/auth';
+import { Auth } from 'aws-amplify';
+
+Auth.configure({
+  // other configurations
+  // ...
+  authenticationFlowType: 'CUSTOM_AUTH',
+});
+
+type SignInParameters = {
+  username: string;
+  password: string;
+};
+
+export async function signIn({ username, password }: SignInParameters) {
+  const challengeResponse = 'the answer for the challenge';
+  try {
+    const user: CognitoUser = await Auth.signIn(username, password);
+    if (user?.challengeName === 'CUSTOM_CHALLENGE') {
+      try {
+        // to send the answer of the custom challenge
+        const challengeAnswerResponse: CognitoUser = await Auth.sendCustomChallengeAnswer(
+          user,
+          challengeResponse
+        );
+        console.log(challengeAnswerResponse);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+  } catch (err) {
+    console.log(err);
+  }
+}
+```
+</Block>
+<Block name="JavaScript">
+
+```js
 import { Auth } from 'aws-amplify';
 
 Auth.configure({
   // other configurations
   // ...
-  authenticationFlowType: 'CUSTOM_AUTH'
+  authenticationFlowType: 'CUSTOM_AUTH',
 });
 
-let challengeResponse = "the answer for the challenge";
-
-Auth.signIn(username, password)
-  .then(user => {
-    if (user.challengeName === 'CUSTOM_CHALLENGE') {
-      // to send the answer of the custom challenge
-      Auth.sendCustomChallengeAnswer(user, challengeResponse)
-        .then(user => console.log(user))
-        .catch(err => console.log(err));
-    } else {
-      console.log(user);
+async function signIn(username, password) {
+  const challengeResponse = 'the answer for the challenge';
+  try {
+    const user = await Auth.signIn(username, password);
+    if (user?.challengeName === 'CUSTOM_CHALLENGE') {
+      try {
+        // to send the answer of the custom challenge
+        const challengeAnswerResponse = await Auth.sendCustomChallengeAnswer(
+          user,
+          challengeResponse
+        );
+        console.log(challengeAnswerResponse);
+      } catch (err) {
+        console.log(err);
+      }
     }
-  })
-  .catch(err => console.log(err));
+  } catch (err) {
+    console.log(err);
+  }
+}
 ```
+</Block>
+</BlockSwitcher>
 
 ### CAPTCHA-based authentication
 
 Here is the sample for creating a CAPTCHA challenge with a Lambda Trigger.
 
 The `Create Auth Challenge Lambda Trigger` creates a CAPTCHA as a challenge to the user. The URL for the CAPTCHA image and  the expected answer is added to the private challenge parameters:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Handler } from 'aws-lambda';
```

Comment:
@abdallahshaban557, do we want to be importing `Handler` here to offer type support for `'aws-lambda'`? Or do we want to do it as `import type {Handler}`?

Reply:
@cwomack I think how it is is fine - https://docs.aws.amazon.com/lambda/latest/dg/typescript-handler.html

Reply:
I agree with Dan as well

Reply:
Thanks @nadetastic for the confirm!